nautobot/dcim/migrations/0044_ensure_virtual_chassis_names_are_unique_and_add_uniqueness_constraint.py

```diff
@@ -0,0 +1,39 @@
+# Generated by Django 3.2.18 on 2023-06-06 21:12
+
+from django.db import migrations, models
+from django.core.exceptions import ValidationError
+
+
+def ensure_virtual_chassis_names_are_all_unique(apps, schema_editor):
+    VC_NAME_TO_COUNT = {}
+    VC = apps.get_model("dcim", "VirtualChassis")
+
+    vcs = VC.objects.all()
+    duplicates = []
+    for vc in vcs:
+        count = VC_NAME_TO_COUNT.setdefault(vc.name, 0)
+        VC_NAME_TO_COUNT[vc.name] = count + 1
+        if count + 1 > 1:
+            duplicates.append(vc.name)
+
+    if duplicates:
+        for duplicate_vc in duplicates:
+            print(f">>> There are {VC_NAME_TO_COUNT[duplicate_vc]} with name {duplicate_vc}\n")
+        print(">>> Unable to proceed with migrations; in Nautobot 2.0+ the name for these records must be unique.\n")
+        print(">>> Failing migration...\n")
+        raise RuntimeError("Duplicate Virtual Chassis record names must be manually resolved before migrating.")
```

Comment:
Take a look at how I'm doing this in #3871 - I *think* my way should be more efficient since it's using a DB annotation for the counts. We should probably standardize these - maybe even create a library function or two.